### ISSUE TITLE
Conky Manager  default widgets and themes have huge issues

### ISSUE THREAD

**Issue by jcklpe** [zcot/conky-manager2#3]
I haven't tried this with any other widgets or themes downloaded from the internet but the defaults that this ships with are having huge issues on my machine. 

Ubuntu 18.04 with Gnome, using Komorebi for my desktop/wallpaper manager. 

Does Conky use GTK because the weird colors seems like maybe they're colors also used by my theme so maybe that's making stuff wonky?

![image](https://user-images.githubusercontent.com/9451726/45247906-c43bdf80-b2d0-11e8-8c07-2846b23aa04e.png)

**Comment by jcklpe** [comment]
Just downloaded a couple of widgets from deviant art and one does not show at all and the other shows fine, though it's spacing etc is still a bit wonky looking. It also disappears behind my desktop wallpaper as soon as I click anything but that has to do with an issue with komorebi. I believe there's a way to fix that with in normal conky but I'm not sure, and I'm also not sure how that crosses over with conky manager etc. 

Also I'm used to using Rainmeter on Windows so maybe this is just not how Conky works etc, but is it not possible to change the position of the widgets by dragging them around? You can with Rainmeter. Also you can access the settings of each widget by simply right clicking on it. Is that not possible with conky?

Here's a pic of how the downloaded one came out:

![image](https://user-images.githubusercontent.com/9451726/45248113-db7bcc80-b2d2-11e8-8847-38351255c008.png)
I got it from here. As you can tell the spacing etc is a mess: https://www.deviantart.com/speedracker/art/Smooth-Conky-V3-Made-for-Conky-Manager-2-2-461461882

**Comment by zcot** [comment]
With conky, yea, you have to mess with it a bit at times to get a particular config to act the way you want it to...

-which is probably one of the reasons why the original conky manager was written. It does offer a quick and easy way to centrally edit a config and restart it(as well as other features like autostart and running multiple configs).

On Ubuntu 18.04 you are running conky 1.10 version if you used the package manager software installer. Most of the configs that are around on various sites are pre-1.10 format. There was a format change for 1.10(among many other things) and although the newer conky will still read the older format, there are certain config variables and calls in "non-basic" configs that do not always act correctly so you end up having to tweak something, or a font is missing, or you want it displayed in a different position/color/format.

This conky manager program has nothing to do with how conky or a config runs. This program is strictly for managing your existing configs against the existing conky software installation. There is no "cross over" or something. conky-manager/2 simply runs the configs for you in a simple non-terminal way for the conky software package that you have installed on the system. It doesn't have a variation of conky within or something. It simply makes the cryptic and long-winded terminal commands for you letting you do it in a GUI setting.

I can't speak about anything specific with Komorebi but it is a very common issue between the different desktop environments that we all have issues with the window type setting. On one desktop it works perfectly, on another you have to change that setting. Or transparency works on one, but you need a different setting if you're using xfce for example. So fixing the disappearing window thing, whether that might be related to Komorebi, MATE, or not, might be something with that own_window_type setting.

A great source of info is here: https://github.com/brndnmtthws/conky/wiki (and specifically the "Settings" and "Variables" pages) *(look in "Settings" page for own_window_type)

From what it sounds like to me, the issues you are experiencing are probably due to configs that don't directly match up to your setup(or configs that are old format and not converting completely properly and/or old and outdated with 3rd party calls, or just not very good).

As for that "Smooth Conky V3 . Made for Conky Manager 2.2" that's a pre-1.10 format and it might not be translating correctly. So it's one you would have to manually tweak to make it right. There is a conversion script available on the conky wiki. Also that config is using some weather thing with a yahooapis.com address that is not functional... -that might be the real problem right there causing a screen formatting issue.

zcot

**Comment by jcklpe** [comment]
Hey thanks, that's great information. I didn't realize that conky manager
was separate from conky, I thought they were connected more tightly. I
guess the solution then is to run the conversion scripts for the pre-1.10
formats.

It was my understanding that conky manager and conky widgets had different
incompatible files? Because from what I've seen all conky manager stuff is
filename.cmb.7z while conky stuff downloads as filename.7z and has to be
unzipped to be used by conky. But is that just a name convention? like if I
unziped a conkyfilename.7z, converted it to the post 1.10 format using the
scripts and zipped it back as conkyfilename.cmb.7z would that work fine
with conky manager 2?

**Comment by zcot** [comment]
Yes, conky-manager and conky-manager2 are just a 3rd party software against conky and its configs. It's there to simply make it easier for a user running conky and its configs.

Be aware though, as for conky itself, that you do not HAVE TO convert any older script. You CAN do it when it is necessary but conky(1.10) itself will natively read the older version file as-is and run it. But sometimes, and it depends on the particular config, it's not 100% good. But largely, you don't HAVE TO convert, because conky automatically does it. If you fire-up conky in a terminal with an older version config you will see it tell you about that, and that it is automatically translating it for you(and at times it might give an error message or useful info about any issues with the config).

for example terminal command: `conky -c ~/.conky/Jess_smooth_3/Smooth_3`
(and [ctrl] + [c] kills it)

If a particular config "doesn't work" and it has to do with "own_window_type" or a transparency or argb setting it's not necessary to convert it, you can just pop open the config in the manager and tweak it and see the changes until you get the right setting.

For this "Import Theme Pack" thing you are talking about, it is apparently just designed to handle *.cmtp.7z files only(I actually don't use that feature and haven't even looked at the source of it). I suspect this came out of the fact that a set of example configs(and support files) were shipped within the conky-manager package and someone decided that should be a naming convention there for a convenience feature.

note: Actually someone could look into changing that because it would be much more useful to be able to open any .7z file or possibly even other compressed formats as well. I've seen plenty of single configs that aren't 7z.. I usually just double-click and there I go extract the thing to ~/.conky and it's good, but it would be a good addition for that feature anyway.

Also to clarify, that archive naming convention has nothing to do with the format of the contained files.

If you want to use that Import feature like that, just rename the 7z file to include the .cmtp.7z ending and conky-manager/2 will see them and do the decompression and file system placement for you and you're good.

If you are absolutely wanting to convert everything you get, then sure, you could unzip - convert - rezip/name. But really, by the time you do that, you can just put the files over in ~/.conky anyway instead of the extra hassle.

Me, I download, double-click, extract(to ~/.conky), then bring up conky-manager2 to run it easily, and if there's an issue then I go digging to find a simple edit for it, or if it's a more advanced config scheme then I'll just convert it if I'm going to be spending any time in it.

zcot

**Comment by corydlamb** [comment]
They all look like this for me as well. Did you ever find a fix?

**Comment by jcklpe** [comment]
Nope. Eventually just gave up.
